opensquirrel/ir.py

```diff
@@ -76,6 +76,13 @@ class Float(Expression):
     def accept(self, visitor: IRVisitor) -> Any:
         return visitor.visit_float(self)
 
+    def __post_init__(self) -> None:
+        try:
+            self.value = float(self.value)
+        except ValueError as exception:
+            msg = "value must be a float"
+            raise ValueError(msg) from exception
+
```

Comment:
I would prefer the following:
```python
def __post_init__(self) -> None:
    if isinstance(self.value, SupportsFloat):
        self.value = float(self.value)
    else:
        msg = "value must be a float"
        raise TypeError(msg)
```
Note that this means that you need to add
```python
from typing import SupportsFloat
```
at the top of the file. (this just checks if `self.value` has a `__float__` method, which is used by `float`.)

Also, the error should be a `TypeError`.